test/chopperbot/message_builder_test.exs

```diff
@@ -0,0 +1,54 @@
+defmodule Chopperbot.MessageBuilderTest do
+  use ExUnit.Case, async: true
+
+  import Mox
+
+  alias Chopperbot.{MessageBuilder, TestMessageBuilder}
+
+  @platform :test
+
+  setup :verify_on_exit!
+
+  describe "build/2" do
+    test "returns error message if the given text is invalid" do
+      text = "splice chopper 100 luffy 200 +v"
+      expected_result = %{text: "error"}
+
+      TestMessageBuilder
+      |> expect(:validate_text_input, fn ^text -> {:error, "invalid format"} end)
+      |> expect(:build_error_message, fn "invalid format" -> expected_result end)
```

Comment:
Wow, the test looks nice. I wanted to try Mox but never have a chance to try.

it seems like the dependencies injection. And looks quite clean.

Reply:
it's a bit painful once the project gets bigger since we have to do DI all the time to make the function testable 😆 

IMO, It's easier to use lib like this https://github.com/edgurgel/mimic

Reply:
Oh, yeah. the usage looks like mocha/expect lib in NodeJS which I like more.